lib/trento/domain/host/host.ex

```diff
@@ -567,4 +631,40 @@ defmodule Trento.Domain.Host do
   def apply(%Host{} = host, %HostRestored{}) do
     %Host{host | deregistered_at: nil}
   end
+
+  defp maybe_emit_host_checks_health_changed_event(
+         %Host{checks_health: checks_health},
+         checks_health
+       ),
+       do: nil
+
+  defp maybe_emit_host_checks_health_changed_event(
+         %Host{host_id: host_id},
+         checks_health
+       ),
+       do: %HostChecksHealthChanged{
+         host_id: host_id,
+         checks_health: checks_health
+       }
+
+  defp maybe_add_checks_health(healths, _, []), do: healths
+  defp maybe_add_checks_health(healths, checks_health, _), do: [checks_health | healths]
+
+  defp maybe_emit_host_health_changed_event(%Host{
+         host_id: host_id,
+         selected_checks: selected_checks,
+         heartbeat: heartbeat,
+         checks_health: checks_health,
+         health: current_health
+       }) do
+    new_health =
+      [heartbeat]
+      |> maybe_add_checks_health(checks_health, selected_checks)
+      |> Enum.filter(& &1)
+      |> HealthService.compute_aggregated_health()
+
+    if new_health != current_health do
+      %HostHealthChanged{host_id: host_id, health: new_health}
+    end
+  end
```

Comment:
This looks a bit convoluted to me, but I noticed we do the same inside the cluster aggregate so upvote for keeping it the same way